lib/rules/no-loss-of-precision.js

```diff
@@ -5,6 +5,209 @@
 
 "use strict";
 
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+/** Class representing a number in scientific notation. */
+class ScientificNotation {
+	/** @type {string} The digits of the coefficient. A decimal point is implied after the first digit. */
+	coefficient;
+
+	/** @type {number} The order of magnitude. */
+	magnitude;
+
+	constructor(coefficient, magnitude) {
+		this.coefficient = coefficient;
+		this.magnitude = magnitude;
+	}
+
+	/* c8 ignore start -- debug only */
+	toString() {
+		return `${this.coefficient[0]}${this.coefficient.length > 1 ? `.${this.coefficient.slice(1)}` : ""}e${this.magnitude}`;
+	}
+	/* c8 ignore stop */
```

Comment:
This isn't necessary for the rule to work. It will format the number in scientific notation when debugging:

<img width="1200" height="333" alt="image" src="https://github.com/user-attachments/assets/991a8f5c-e35d-4d5d-853c-2bde8c6a97ed" />